microsoft-edge/progressive-web-apps-chromium/how-to/offline.md

```diff
@@ -1,183 +1,155 @@
-                    return response;
-                })
-                // If the network is unavailable to make a request, pull the offline
-                // page out of the cache.
-                .catch(() => caches.match( "/offline/" ));
-            })
-        ); // end respondWith
-    } // end if HTML
-});
+const browserInformation = {
+  name: 'Microsoft Edge',
+  version: 108
+};
+
+localStorage.setItem('browser', JSON.stringify(browserInformation));
 ```
 
-The above code runs within the Service Worker whenever the browser makes a `fetch` request for this site. Within that event, there is a conditional statement that runs if the request is for an HTML file. The Service Worker checks to see whether the file already exists in any cache, by using the `match()` method:
+The above code stores a JavaScript object as a JSON string in `localStorage` using the `setItem()` method, and assigns a key equal to `browser`. You can retrieve the information from `localStorage` by using the `getItem()` method as shown below:
 
-*  If the request exists in the cache, that cached result is returned.
-*  If the request doesn't exist in the cache, a new `fetch` for that resource is run, a copy of the response is cached for later, and the response is returned.
-   * If the `fetch` fails because the network is unavailable, the offline page is returned from the cache.
+```javascript
+const value = localStorage.getItem('browser');
+
+const browserInformation = JSON.parse(value);
+```
 
-This simple introduction shows how to use caching in your progressive web app (PWA). Each PWA is different and may use different caching strategies. Your code may look different, and you can use different caching strategies for different routes within the same application.
+To learn more, see [Web Storage API](https://developer.mozilla.org/docs/Web/API/Web_Storage_API) on MDN.
 
 
 <!-- ====================================================================== -->
-## Use IndexedDB in your PWA to store structured data
+## IndexedDB
 
-`IndexedDB` is an API for storing structured data. Similar to the `Cache` API, it's also asynchronous. This means you can use it in the main thread, or with Web Workers such as Service Workers. Use the `IndexedDB` API for storing a significant amount of structured data on the client, or binary data, such as encrypted media objects.  See [MDN primer on using IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API/Using_IndexedDB).
+IndexedDB is an asynchronous API for storing structured data that can be used in your app's front-end code or service worker code. Use the IndexedDB API for storing a significant amount of structured data on the client, or binary data, such as encrypted media objects or files.
 
+IndexedDB is the best option for storing data in your PWA because using the API does not slow down your app by blocking the main thread, and it can be used both from your app's front-end code and service worker.
```

Comment:
```suggestion
IndexedDB is the best option for storing data in your PWA, because using the API doesn't slow down your app by blocking the main thread, and it can be used both from your app's front-end code and service worker.
```